src/v/coproc/reconciliation_backend.cc

```diff
@@ -25,12 +27,18 @@ namespace coproc {
 reconciliation_backend::reconciliation_backend(
   ss::sharded<cluster::topic_table>& topics,
   ss::sharded<cluster::shard_table>& shard_table,
-  ss::sharded<storage::api>& storage) noexcept
+  ss::sharded<cluster::partition_manager>& cluster_pm,
+  ss::sharded<partition_manager>& coproc_pm) noexcept
   : _self(model::node_id(config::node().node_id))
   , _data_directory(config::node().data_directory().as_sstring())
   , _topics(topics)
   , _shard_table(shard_table)
-  , _storage(storage) {}
+  , _cluster_pm(cluster_pm)
+  , _coproc_pm(coproc_pm) {
+    _retry_timer.set_callback([this] {
+        (void)within_context([this]() { return fetch_and_reconcile(); });
```

Comment:
should fetch_and_reconcile reset the timer instead of recursing?

Reply:
The timer is fired on a condition, this isn't supposed to loop back around